docs/bots-guide.md

```diff
@@ -0,0 +1,192 @@
+5. Subscribe the bot to any streams that the bot needs to read message from or write message to.55
+ * To subscribe your bot to streams, navigate to *Manage Streams*. Select a stream and add your bot by its email address (the address you assigned in step 2).
+ * Now, the bot will do its job on the streams you subscribed it to.
+6. Run the bot.
+ * On your Zulip server, navigate to `~/zulip/contrib_bots/`
+ * Run `python run.py ~/zulip/contrib_bots/<my-bot>.py --config-file ~/.zuliprc
+ * Check the output of the command. It should start with the text the `usage` function returns, followed by logging output similar to this:
+
+>INFO:root:starting message handling...
+>INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): localhost
+
+* Congrats! Now, your bot should be ready to test on the streams you subscribed it to.
+
+### Test the `followup.py` bot
+1. Do the previous steps for the `followup.py` bot.
+2. Subscribe the bot to *followup* and another stream, e.g. *social*.
+3. Send a message to the stream *social* . If everything works, a copy of this message should now pop up in the *followup* stream.
+
+## How to develop a bot
+
+Below explains the structure of a bot `<my-bot>.py`. You can use this as boilerplate code for developing your own bot.
+
+Every bot is built upon this structure:
+```
+class FollowupHandler(object):
+    '''
+    A docstring documenting this bot.
+    '''
+
+    def usage(self):
+        return '''Your description of the bot'''
+
+    def triage_message(self, message):
+        #add your code here
+
+    def handle_message(self, message, client, state_handler):
+# add your code here
+
+handler_class = FollowupHandler
+```
+* The class name (in this case *FollowupHandler*) can be defined by you and should match the name of your bot. To register your bot's class, adjust the last line `handler_class = FollwupHandler` to match your class name.
+* Every bot needs to implement the functions
+ * `usage(self)`
+ * `triage_message(self, message)`
+ * `handle_message(self, message, client)`
+* These functions are documented in the [next section](#bot-API).
+
+## Bot API
+This sections documents the functions every bot needs to implement and the structure of the bot's config file.
+
+### usage(self)
+Is called by the system to obtain information about this bot.
+#### Arguments
+* self - the instance the method is called on.
+#### Return values
+* A string describing the bot's functionality
+#### Example implementation
+```
+def usage(self):
+    return '''
+        This plugin will allow users to flag messages
+        as being follow-up items.  Users should preface
+        messages with "@followup".
+        Before running this, make sure to create a stream
+        called "followup" that your API user can send to.
+        '''
+```
+
+### triage_message(self, message)
+Is called by the system when a message was sent.
+#### Arguments
+* self - the instance the method is called on.
```

Comment:
Superfluous full stop.

I'd also add more spacing between the `*` here to improve readability.